### ISSUE TITLE
Some icons in "Default" theme are needlessly overridden, breaking doom themes

### ISSUE THREAD

**Issue by eeshugerman** [emacs-lsp/lsp-treemacs#89]
Original issue: https://github.com/hlissner/emacs-doom-themes/issues/500

@nbfalcon's assessment: https://github.com/emacs-lsp/lsp-treemacs/issues/66#issuecomment-761086890

> So the issue here is that we are modifying the "Default" treemacs theme, and replacing some icons which were probably provided by all-the-icons with png icons. The former uses a font and so they are vector graphics and scale well, while the latter ones are raster graphics and don't. The icons that are needlessly overridden in lsp-treemacs-icons would need to be examined. Note that this is unrelated to this issue, though.

**Comment by eeshugerman** [comment]
> The icons that are needlessly overridden in lsp-treemacs-icons would need to be examined

@nbfalcon so is this just a matter of comparing the ["Default" icon list in lsp-treemacs](https://github.com/emacs-lsp/lsp-treemacs/blob/master/lsp-treemacs-themes.el#L34) vs that [in treemacs](https://github.com/Alexander-Miller/treemacs/blob/master/src/elisp/treemacs-icons.el#L267)? not sure i'm on the right track.

**Comment by nbfalcon** [comment]
That is a start. However, we must also consider `all-the-icons`, which might have additional icons.

**Comment by eeshugerman** [comment]
hmm, might be on the wrong track here. it seems my test of your triage branch was flawed: if i checkout upstream/master the issue is still "resolved", so the commented-out `treemacs-modify-theme` form doesn't seem to be what's making the difference.

rather it would seem to have something to do with how i'm installing `lsp-treemacs` from my git clone, which is by adding `(package-install-file "~/devel/misc/lsp-treemacs")` in my `dotspacemacs/user-config`, just before `(doom-themes-treemacs-config)`.

🤔 i'll keep experimenting

**Comment by seagle0128** [comment]
Same issue here. I think we should not modify "Default" theme directly. Creating a new one will avoid the conflicts.

**Comment by nbfalcon** [comment]
@eeshugerman have you reloaded `lsp-treemacs-themes`?

**Comment by eeshugerman** [comment]
reloading the library after the issue is visible does nothing, but if i do `(load-library "lsp-treemacs-themes")` in my config (where i was reinstalling before), the issue never appears.

that's actually a pretty pleasant workaround! :)

**Comment by seagle0128** [comment]
My workaround is 
```elisp
(with-eval-after-load 'lsp-treemacs
          (doom-themes-treemacs-config))
```

**Comment by ig-perez** [comment]
Same issue with a vanilla Emacs installation:
![image](https://user-images.githubusercontent.com/33359427/107094284-bdcd2d00-67d4-11eb-88cc-16d2caed39cf.png)
`LICENSE` and `poetry.lock` icons are too big and moved to the left.

I guess previous workarounds are for Doom Emacs, I have tried both workarounds and didn't help.

Thanks

**Comment by seagle0128** [comment]
@ig-perez The workaround is loading `doom-themes-ext-treemacs` after `lsp-treemacs`. Try evaluating the buffer of doom-themes-ext-treemacs.

**Comment by jul1u5** [comment]
I tried @seagle0128 workaround on Doom Emacs and it didn't seem to help.
> ```emacs-lisp
> (with-eval-after-load 'lsp-treemacs
>           (doom-themes-treemacs-config))
> ```

However, it seems the following worked:
```emacs-lisp
(after! lsp-treemacs
  (load-library "doom-themes-ext-treemacs"))
```

**Comment by ig-perez** [comment]
Thanks @seagle0128 I tried:
```emacs-lisp
(with-eval-after-load 'lsp-treemacs (doom-themes-ext-treemacs))
```
But I got: `Error processing message (void-function doom-themes-ext-treemacs)`

@jul1u5 are you under Doom right?

**Comment by jul1u5** [comment]
@ig-perez Yes, I am using Doom and I think I got the same error with that workaround. Have you tried using the load-library one?

**Comment by ig-perez** [comment]
Hey @jul1u5 I tried and well, in some way it solved the problem by disappearing the icons:
![image](https://user-images.githubusercontent.com/33359427/107689506-e80c6800-6c76-11eb-9dc8-2b7b2de7084c.png)

So, I'm still with this issue.

**Comment by jul1u5** [comment]
Sorry, I don't see what's wrong here. Should .properties or .lock files have specific icons?

It looks the same for me, I thought that's how the icons are supposed to look like 😀

**Comment by ig-perez** [comment]
Yeah, these are the icons:
![image](https://user-images.githubusercontent.com/33359427/107699167-ad5cfc80-6c83-11eb-92c3-a2760bf9aa0f.png)

They look good, the only thing is the size/alignment. They should be the same size and at the same level of `pyproject.toml` :)

**Comment by seagle0128** [comment]
> Thanks @seagle0128 I tried:
> 
> ```emacs-lisp
> (with-eval-after-load 'lsp-treemacs (doom-themes-ext-treemacs))
> ```
> 
> But I got: `Error processing message (void-function doom-themes-ext-treemacs)`
> 
> @jul1u5 are you under Doom right?

My configurations:
```
(use-package doom-themes
        :custom-face
        (doom-modeline-buffer-file ((t (:inherit (mode-line bold)))))
        :custom
        (doom-themes-treemacs-theme "doom-colors")
        :init (centaur-load-theme centaur-theme t)
        :config
        ;; Enable flashing mode-line on errors
        (doom-themes-visual-bell-config)

        ;; Enable customized theme
        ;; FIXME https://github.com/emacs-lsp/lsp-treemacs/issues/89
        (with-eval-after-load 'lsp-treemacs
          (doom-themes-treemacs-config)))
```

**Comment by ig-perez** [comment]
Thanks @seagle0128 , all of this is confusing. Let me try to summarize.

`treemacs` default theme looks like this:
![image](https://user-images.githubusercontent.com/33359427/108582179-76c85700-72ff-11eb-941d-762184eb66e3.png)

The yellow folder icons are dissonant with `doom-themes`, so I guess it is better to enable the configuration for `treeemacs` with `(doom-themes-treemacs-config)` from the beggining, not only after LSP.

But when I do that, the icons look unaligned as shown initially:
![image](https://user-images.githubusercontent.com/33359427/108582566-2d2d3b80-7302-11eb-90d8-0d763341dde1.png)
![image](https://user-images.githubusercontent.com/33359427/108582590-5cdc4380-7302-11eb-9def-1ba99d090165.png)

The proposed fix will keep original `treemacs` theme and just activates the  `doom-themes` after LSP mode. This is odd since I use `treemacs` not only for LSP, I'd prefer a smother experience keeping the same `doom-themes` look and feel wherever I go in Emacs. Moreover it seems it is replacing the original `treemacs` icons with smaller ones:
![image](https://user-images.githubusercontent.com/33359427/108582662-ae84ce00-7302-11eb-99bf-c934a19ab60c.png)


So the thing is (IMO):
- Enable the doom-theme for treemacs from the beginning, not only after LSP
- Keep the cool `treemacs` icons  (`all-the-icons` I guess)

Now I'm doubtful, is this an issue with `treemacs` or `doom-themes`. Which package is replacing the `treemacs` icons?

**Comment by seagle0128** [comment]
@ig-perez The fact is if disabling or uninstalling `lsp-treemacs`, all icons are correct. There should be some conflicts.
